explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/GroupStatisticalParityDifferenceEndpoint.java

```diff
@@ -65,7 +61,13 @@ public Response spd(BaseMetricRequest request) throws DataframeCreateException {
 
         final Dataframe df = dataSource.getDataframe();
 
-        final double spd = calculator.calculateSPD(df, request);
+        final double spd;
+        try {
+            spd = calculator.calculateSPD(df, request);
+        } catch (MetricCalculationException e) {
+            LOG.error("Error calculating metric: " + e.getMessage());
```

Comment:
Let's log the stacktrace for debugging
```suggestion
            LOG.error("Error calculating metric: " + e.getMessage(), e);
```